### DISCUSSION TITLE
Make fullscreen the first default layout

### DISCUSSION THREAD

**Discussion by gmayerw** [xmonad/xmonad#356] [Q&A] [answered]
How would I make the default layout on every workspace fullscreen? I would still like to switch between horizontal and vertical layouts using the $mod + space button

**Comment by geekosaur** [comment, marked as answer]
The default `layoutHook` is

    layoutHook = tiled ||| Mirror tiled ||| Full

where `tiled` is defined elsewhere (but you'll probably need to copy it into your config since it's not exported). You would rearrange this list so `Full` is first.

For completeness, the definition of `tiled` is

    tiled = Tall 1 (3/100) (1/2)

**Reply by gmayerw** [reply]
Here is my xmonad.hs:

```
import XMonad

main = xmonad defaultConfig
        { modMask = mod4Mask
        , terminal = "xterm -e vifm -c empty /home/wgm"
        , borderWidth = 0
        }
```

Would I put `layoutHook = Full ||| tiled ||| Mirror tiled` in line like this?

```
import XMonad

main = xmonad defaultConfig
        { modMask = mod4Mask
        , terminal = "xterm -e vifm -c empty /home/wgm"
        , borderWidth = 0
        , layoutHook = Full ||| tiled ||| Mirror tiled
        }
```

Thanks for your answer!

**Reply by geekosaur** [reply]
Yes, with the definition of `tiled` after the close brace (at the same indentation level as `main`).

**Comment by gmayerw** [comment]
Thanks! That worked! My final code was:

```
import XMonad

main = xmonad defaultConfig
        { modMask = mod4Mask
        , terminal = "xterm -e vifm -c empty /home/wgm"
        , borderWidth = 0
        , layoutHook = Full ||| tiled ||| Mirror tiled
        }

tiled = Tall 1 (3/100) (1/2)
```

**Comment by gmayerw** [comment]
Following up on this @geekosaur - how could I do this for this hook:
```
  , layoutHook = hiddenWindows $ noBorders $ avoidFloats $ toggleLayouts Full
               $ spacingRaw False (Border 0 0 0 0) True (Border 0 0 0 0) True
               $ ifMax 2 (Mirror Grid) Grid
```

I tried moving `toggleLayouts Full` before `hiddenWindows` but then some keys weren't working so I assume that broke something.

**Comment by geekosaur** [comment]
I don't see how that would affect keys; they're a completely separate hook. Which keys weren't working?

**Comment by gmayerw** [comment]
It was the `mod4 + z` that was not working but looks like it actually is working. Must have been something else causing that. Now I have:
```
  , layoutHook = toggleLayouts Full $ hiddenWindows $ noBorders $ avoidFloats $ spacingRaw False (Border 0 0 0 0) True (Border 0 0 0 0) True $ ifMax 2 (Mirror Grid) Grid
  , borderWidth = 0
  , workspaces = ["1"]
  , modMask = mod4Mask
  , terminal = "kitty"
  , logHook = sinkAllLogHook
  } `additionalKeys`
  [ ((mod4Mask, xK_space), sendMessage ToggleStruts >> sendMessage (Toggle "Full"))
  , ((mod4Mask, xK_z), withFocused hideWindow)
```

For some reason it still will not launch in full screen mode

**Comment by geekosaur** [comment]
It won't; `toggleLayouts Full` doesn't start toggled on, it's just a quick way to switch to `Full` and back.

**Comment by gmayerw** [comment]
Got it! So to start in full do I just type Full first?

**Comment by geekosaur** [comment]
Yes. There are some other things but they're strictly hacks and in general more complicated than just putting the layouts in the order you want. If you want different orderings on some workspaces consider [`XMonad.Layout.PerWorkspace`](https://hackage.haskell.org/package/xmonad-contrib-0.18.1/docs/XMonad-Layout-PerWorkspace.html).

**Comment by gmayerw** [comment]
I think I am looking for the same behavior as I had in the original
question which is just to start in fullscreen mode but having issues with
the code and syntax now that I have a more advanced layoutHook

**Comment by gmayerw** [comment]


**Reply by geekosaur** [reply]
That was what I was hinting at with my comment about more complex solutions, but yes, the simplistic approach to it won't work and it's mostly not worth the more complex versions that do.

Also, ChatGPT knows virtually nothing about Haskell, much less xmonad, and its suggestions in that area are largely worthless.

**Comment by geekosaur** [comment]
```
    layoutHook = hiddenWindows $ noBorders $ avoidFloats $ Full ||| (spacingWithEdge 4 $ ifMax 2 (Mirror Grid) Grid)
```

**Comment by gmayerw** [comment]
Thank you - I tried this but didn't have the parenthesis. Noted

**Comment by gmayerw** [comment]
Agreed on ChatGPT. It is not Haskell fluent
